addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -14299,4 +14300,77 @@ QUnit.module("Views", (hooks) => {
         await editInput(target, "[name='int_field'] input", "42");
         assert.verifySteps(["get_special_data 9", "get_special_data 42"]);
     });
+
+    QUnit.test(
+        "reload records in the context of the form to avoid having partial field values",
+        async function (assert) {
+            serverData.actions = {
+                1: {
+                    id: 1,
+                    name: "Partner",
+                    res_model: "partner",
+                    type: "ir.actions.act_window",
+                    views: [[false, "form"]],
+                    view_mode: "form",
+                    res_id: 6,
+                    target: "new",
+                },
+            };
+
+            serverData.views = {
+                "partner,false,form": `<form>
+                <field name="display_name"/>
+                <field name="timmy" widget="many2many_tags" options="{'color_field': 'color'}"/>
+            </form>`,
+            };
+
+            await makeView({
+                type: "form",
+                resModel: "user",
+                resId: 19,
+                serverData,
+                arch: `
+            <form>
+                <field name="partner_ids">
+                    <tree>
+                        <field name="display_name"/>
+                    </tree>
+                    <form>
+                        <header>
+                            <button type="action" name="1" string="test"/>
+                        </header>
+                        <field name="display_name"/>
+                        <field name="timmy" widget="many2many_tags" options="{'color_field': 'color'}"/>
```

Comment:
options not necessary here

Reply:
actually, is the many2many necessary? Wouldn't it be still relevant if you simply edit the display_name?